modules/dists/BlockDist.chpl

```diff
@@ -1535,6 +1535,14 @@ where this.sparseLayoutType == unmanaged DefaultDist &&
   return true;
 }
 
+// Block1 <=> Block2 
+proc BlockArr.doiSwap(arr) {
+  coforall (loc, localArr1, localArr2) in zip(Locales,this.locArr, arr.locArr) {
```

Comment:
I think you forgot an `on loc` here. So, your performance will be even better when you do that :)

Reply:
And, if you do that, `BlockArr` has `myLocArr` that is a reference to the local item in `locArr`. (This is established by privatization) So, you can just do:

```chapel
coforall loc in Locales do on loc {
  // myLocArr.myElems.....
}
```